galasa-managers-parent/galasa-managers-zos-parent/dev.galasa.zos3270.manager/src/main/java/dev/galasa/zos3270/internal/gherkin/Gherkin3270GivenTerminal.java

```diff
@@ -54,17 +61,79 @@ public void allocateTerminal(IGherkinExecutable executable, Map<String,Object> t
 
     public void provision(IGherkinExecutable executable) throws Zos3270ManagerException {
         List<String> groups = executable.getRegexGroups();
+
+        // Extract the parameters from the step in the scenario.
         String terminalId = Gherkin3270Coordinator.defaultTerminaId(groups.get(1));
         String imageTag = Gherkin3270Coordinator.defaultImageTag(groups.get(3));
+        String rowsStepParameter = groups.get(5);
+        String columnsStepParameter = groups.get(6);
+
+        // Log what we have collected.
+        {
+            StringBuffer msg = new StringBuffer();
+            msg.append("Provisioning a terminal:");
+            msg.append(" id=");
+            msg.append(terminalId);
+            msg.append(" imageTag=");
+            msg.append(imageTag);
+            msg.append(" rows=");
+            msg.append(rowsStepParameter);
+            msg.append(" columns=");
+            msg.append(columnsStepParameter);
+            logger.info(msg.toString());
+        }
         
         Zos3270TerminalImpl newTerminal = this.gerkinCoordinator.getTerminal(terminalId);
         if (newTerminal == null) {
-            TerminalSize terminalSize = new TerminalSize(80, 24);
+            TerminalSize terminalSize = getPreferredTerminalSize(rowsStepParameter , columnsStepParameter);
             TerminalSize alternateSize = new TerminalSize(0, 0);
 
             newTerminal = this.manager.generateTerminal(imageTag, true, terminalSize, alternateSize);
             this.gerkinCoordinator.registerTerminal(terminalId, newTerminal, imageTag);
             logger.info("zOS 3270 Terminal id '" + terminalId + "' as been provisioned for image tag '" + imageTag + "'");
         } 
     }
+
+    protected TerminalSize getPreferredTerminalSize(String rowsStepParameter, String rowsColumnParameter) throws Zos3270ManagerException {
+        int columns ;
+        int rows ;
+        rows = getNumericCPSProperty("zos3270.gherkin.terminal.rows",DEFAULT_TERMINAL_ROWS,rowsStepParameter);
+        columns = getNumericCPSProperty("zos3270.gherkin.terminal.columns",DEFAULT_TERMINAL_COLUMNS,rowsColumnParameter);
+        logger.info("Preferred terminal size is "+Integer.toString(rows)+" x "+Integer.toString(columns));
```

Comment:
Rows and columns might need swapping for this log message - for example with 24 rows and 80 columns, this would read as `24 x 80` instead of `80 x 24`

Reply:
I do wonder whether mainframers use 24x80 

This is from host on demand:
<img width="609" alt="image" src="https://github.com/user-attachments/assets/0fd6a261-40ed-4c8e-85f4-3b1eeef8d9e9">

Reply:
Adam Coulthard
> Its normally 24 x 80 rows and columns